### ISSUE TITLE
Combobox multi-select improvements

### ISSUE THREAD

**Issue by gretanausedaite** [iTwin/iTwinUI#881]
Combobox multi-select functionality has some concerns that could be improved:
- researched the accessibility concerns that come with this markup
- fix focus when multi-select is enabled
	- Scroll to one of the selected items on menu open
- test performance - we have some foreach loops that could cause problems with bigger datasets

**Comment by mayank99** [comment]
Adding some comments here as I've been working a lot with ComboBox multiselect lately.

In response to the original issue:
- Accessibility of multi-select has been improved in #1228, and it also unblocks these two related issues:
  - #894 
  - #907
- Focus still needs to be fixed. :(
- Performance seems to be ok? Haven't seen any user complaints so far, and virtualization is available too.

One new issue I noticed: after typing in a combobox and selecting an option (using <kbd>Enter</kbd>), I would have expected the text field to be cleared.

Also, when testing combobox in VO/NVDA (in general, not just multiselect), I was noticing weird behaviors and having some difficulty opening the menu. Will need to test and research some more more. It might be better UX if we add inline autocomplete (#862).

**Comment by jfisher-bentley** [comment]
@mayank99 the issue with selecting an option and the text field not being cleared is also an issue if you select with the mouse (not just Enter). Should I create a separate issue specifically for that? (users are complaining about that issue specifically within our UI)

**Comment by mayank99** [comment]
> It is still reproducible on the docs site, so I assumed it was still an issue (https://itwinui.bentley.com/docs/combobox)

The latest version of the docs hasn't been deployed yet. In the meantime, you can look at the DEV version: https://dev-itwinui.bentley.com/docs/combobox

**Comment by FlyersPh9** [comment]
iTwinUI is now in maintenance mode per [the support policy](https://github.com/iTwin/iTwinUI/wiki/Support-policy), with [StrataKit](https://github.com/iTwin/stratakit) replacing it as the primary design system.
